### PR TITLE
Rebased and addressing issue 141 and 135

### PR DESCRIPTION
**Pull request by Surax98**
# Summary

Rebased the branch on the latest InterLink main upstream
Addressing issue 141: now the VK is not stuck anymore if the Pod is being deleted while waiting for a ConfigMap/Secret
Also improved the get status, both VK and SLURM Sidecar side

**Related issue :**
#135 #141

### CODE REVIEW

File: pkg/virtualkubelet/virtualkubelet.go
Comment (dciangot) [inline pkg/virtualkubelet/virtualkubelet.go:1]: are we waiting for this coroutines somewhere? is there a reason for them? Do we risk to accumulate routines indefenetly?

**Comment by dciangot** [comment]
- [x] check my comment in the PR
- [x] test log retrieval in slurm sidecar and check that nothing break
- [x] test all examples and see that nothing break
- [x] include example for k8s deployment test
- [x] try failure message in case of interlnk remote down

@Surax98  task before accepting

File: pkg/virtualkubelet/virtualkubelet.go
Comment (Surax98) [reply pkg/virtualkubelet/virtualkubelet.go:1]: Yes, at the moment it is actually possible the coroutines get stuck, forever waiting for a configmap or a secret. I am currently adding a timer to return a timeout error, eventually

**Review by dciangot — COMMENTED** [review]
LGTM

**Review by dciangot — APPROVED** [review]

**Pull request merged**